Bump mono to the latest of 2018-08

mandel-macaque opened the pull request:
Includes the following commits:

* Fix Windows MSI build, msbuild is now in Current instead of 15.0 https://github.com/mono/mono/commit/d1b7bf2c472e32d174610b776c8800e436423457
* More Windows MSI fixes https://github.com/mono/mono/commit/ea8ae3e8e05be7729d3278b943e612e3850a2877
* Fix Windows MSI build https://github.com/mono/mono/commit/18fde9f8c29b96defb3b04e4093cdb1d16595eee
* Bump msbuild to track mono-2018-08 https://github.com/mono/mono/commit/082e1a23463e52b611b1cbc12ae86af6b2f376bf
* Bump msbuild to track mono-2018-08 https://github.com/mono/mono/commit/167f04c9024a1ecb0a9bf5865e05e352a41b584a
* A deadlock happens when during a process_suspend there is a pending invoke and then when it's calling the invoke there is another call to process_suspend in the same thread.https://github.com/mono/mono/commit/6843da5c570ebf84495dde7b1670388321839f86
* [Facades] Include missing enums in System.Memory https://github.com/mono/mono/commit/913c11def52c288c0fe615a8eda5847680f90f24
* Bump API snapshot submodule https://github.com/mono/mono/commit/223ea7ef92eb396f1570924cf401f931be778f50
* [2018-08] [arm64] Correct exception filter stack frame size (#14769) https://github.com/mono/mono/commit/3a07bd426d3dae0d9b6c7891e7bb11f487b18484

Complete diff: https://github.com/mono/mono/compare/3cb36842fc47eb8d749cabe351b3902c24065d77...3a07bd426d3dae0d9b6c7891e7bb11f487b18484

monojenkins commented:
Build success
ℹ️ [Skipped execution](https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/10213//console)

mandel-macaque commented:
lets wait for the internal tests to be done

mandel-macaque commented:
@spouliot @rolfbjarne zip failures are due to files missing in the tests dlls, something that was added in the later mono test dlls. I'll ignore them in this branch. The https tests. is due to the network :/

mandel-macaque commented:
Added the required ignores. The issue is that the required resources are not added in the test dlls. This just happens in this branch because we are not using the mono sdk for the mac tests, something that does work in other branches. 

I added the ignore to get cleaner test results, does not affect the customers.

monojenkins commented:
Build success
ℹ️ [Skipped execution](https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/10225//console)

monojenkins commented:
Build success
ℹ️ [Skipped execution](https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/10230//console)

mandel-macaque commented:
Issues are known and fixed in the backport PR: https://github.com/xamarin/xamarin-macios/pull/6266
